opentelemetry-proto/src/transform/logs.rs

```diff
@@ -98,14 +98,22 @@ pub mod tonic {
                             }),
                         })
                         .collect();
-                    if let Some(event_name) = log_record.event_name.as_ref() {
-                        attributes.push(KeyValue {
-                            key: "name".into(),
-                            value: Some(AnyValue {
-                                value: Some(Value::StringValue(event_name.to_string())),
-                            }),
-                        });
+                    #[cfg(feature = "populate-logs-event-name")]
```

Comment:
I think it would be simpler to use `cfg!` macro instead of the attribute here:

```rust
let mut attributes: Vec<KeyValue> = log_record
    .attributes_iter()
    .map(|kv| KeyValue {
        key: kv.0.to_string(),
        value: Some(AnyValue {
            value: Some(kv.1.clone().into()),
        }),
    })
    .collect();

if cfg!(feature = "populate-logs-event-name") {
    if let Some(event_name) = &log_record.event_name {
        attributes.push(KeyValue {
            key: "name".into(),
            value: Some(AnyValue {
                value: Some(Value::StringValue(event_name.to_string())),
            }),
        });
    }
}
attributes
```

Reply:
I think `cfg!` macro has runtime overhead, so the code (both if/else branch) would be part of the compiled binary.